tests/test_ecommerce_api.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+"""
+Tests for the `edx-enterprise` Ecommerce API module.
+"""
+
+from __future__ import absolute_import, unicode_literals, with_statement
+
+
+import unittest
+
+import mock
+
+from django.contrib.auth.models import User
+
+from enterprise.ecommerce_api import EcommerceApiClient
+
+
+class TestEcommerceApiClientInitialization(unittest.TestCase):
+    """
+    Test the setup process for building an EcommerceApiClient instance.
+    """
+
+    @mock.patch('enterprise.ecommerce_api.ecommerce_api_client')
+    def test_init_not_configured(self, _):
+        client = EcommerceApiClient(mock.MagicMock(spec=User))
+        assert client.client is None
+
+    def test_init_no_client(self):
+        client = EcommerceApiClient(mock.MagicMock(spec=User))
+        assert client.client is None
+
+    @mock.patch('enterprise.ecommerce_api.ecommerce_api_client')
+    @mock.patch('enterprise.ecommerce_api.is_commerce_service_configured')
+    def test_init_configured(self, mock_is_configured, mock_api_client):
+        mock_is_configured.return_value = True
+        resulting_api_client_singleton = object()
+        mock_api_client.return_value = resulting_api_client_singleton
+        client = EcommerceApiClient(mock.MagicMock(spec=User))
+        assert client.client is resulting_api_client_singleton
+
+
+class TestEcommerceApiClientBehavior(unittest.TestCase):
+    """
+    Test the behavior of getting a single coupon by ID.
```

Comment:
@haikuginger Let's go with a more general docstring here, to take into account the fact that this class also tests behavior for getting multiple coupons.